Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/securitycontextconstraints/provider.go

```diff
@@ -0,0 +1,176 @@
+	}, nil
+}
+
+// Create a SecurityContext based on the given constraints.  If a setting is already set on the
+// container's security context then it will not be changed.  Validation should be used after
+// the context is created to ensure it complies with the required restrictions.
+func (s *simpleProvider) CreateSecurityContext(pod *api.Pod, container *api.Container) (*api.SecurityContext, error) {
+	var sc *api.SecurityContext = nil
+	if container.SecurityContext != nil {
+		// make a copy so we aren't working on the actual container data.  This is for safety
+		// so that if this is called directly (ie. not in a loop which uses a copy anyway)
+		// we aren't changing a container which would cause subsequent calls to this method
+		// to not change anything since values are already set.
+		scCopy, err := conversion.DeepCopy(container.SecurityContext)
+		if err != nil {
+			return nil, err
+		}
+		var ok bool
+		sc, ok = scCopy.(*api.SecurityContext)
+		if !ok {
+			return nil, fmt.Errorf("Unable to use copied security context")
+		}
+	} else {
+		sc = &api.SecurityContext{}
+	}
+	if sc.RunAsUser == nil {
+		uid, err := s.runAsUserStrategy.Generate(pod, container)
+		if err != nil {
+			return nil, err
+		}
+		sc.RunAsUser = uid
+	}
+
+	if sc.SELinuxOptions == nil {
+		seLinux, err := s.seLinuxStrategy.Generate(pod, container)
+		if err != nil {
+			return nil, err
+		}
+		sc.SELinuxOptions = seLinux
+	}
+
+	if sc.Privileged == nil {
+		priv := false
+		sc.Privileged = &priv
+	}
+
+	// No need to touch capabilities, they will validate or not.
+	return sc, nil
+}
+
+// Ensure a container's SecurityContext is in compliance with the given constraints
+func (s *simpleProvider) ValidateSecurityContext(pod *api.Pod, container *api.Container) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+
+	if container.SecurityContext == nil {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("securityContext", container.SecurityContext, "No security context is set"))
+		return allErrs
+	}
+
+	sc := container.SecurityContext
+	allErrs = append(allErrs, s.runAsUserStrategy.Validate(pod, container)...)
+	allErrs = append(allErrs, s.seLinuxStrategy.Validate(pod, container)...)
+
+	if !s.scc.AllowPrivilegedContainer && *sc.Privileged {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("privileged", *sc.Privileged, "Privileged containers are not allowed"))
+	}
+
+	if sc.Capabilities != nil && len(sc.Capabilities.Add) > 0 {
+		for _, cap := range sc.Capabilities.Add {
+			found := false
+			for _, allowedCap := range s.scc.AllowedCapabilities {
+				if cap == allowedCap {
+					found = true
+					break
+				}
+			}
+			if !found {
+				allErrs = append(allErrs, fielderrors.NewFieldInvalid("capabilities.add", cap, "Capability is not allowed to be added"))
+			}
+		}
+	}
+
+	if !s.scc.AllowHostDirVolumePlugin {
+		// TODO: the volume is not yet available if we validate in admission, how can we validate this?
```

Comment:
You want to look for:

1.  Whether `pod.Volumes` contains a `hostPath` volume (ie, volume.HostPath)
2.  Whether `pod.Volumes` contains a volume with a persistent volume claim source that refers to a `hostPath` persistent volume.

Btw, `scc.AllowHostDirVolumePlugin` should be `scc.AllowHostPathVolumePlugin`